sdk/include/opentelemetry/sdk/metrics/view/attributes_processor.h

```diff
@@ -18,8 +19,124 @@ namespace sdk
 {
 namespace metrics
 {
+
 using MetricAttributes = opentelemetry::sdk::metrics::FilteredOrderedAttributeMap;
 
+/**
+ * Generic FNV hash implementation
```

Comment:
The code for this new hash function should be in a separate header file.

Also, some pointers to a paper explaining the magic behind it would be welcome.

Right now the code just contains magic values, and can not be maintained.